MobilityData/mobility-feed-api
# feat: 248 add load test to GitHub actions

**Summary:**

Closes #248 

**Expected behavior:** 

- added load test documentation
- schedule a load test cron job every other friday
- start a load test upon success of deployment to qa

**Testing tips:**

The easiest way to test would be to start the action manually, but this is not possible until the branch is merged into master.
A workaround to do this would be to create your own branch from `248-add-load-test-to-github-actions` (e.g `MY_BRANCH`) and add this as a trigger in `.github/workflows/schedule-load-test.yml`:
```
  push:
    branches: [ MY_BRANCH ] 
```
After the action runs, you can examine the artifact.
After the testing, just delete `MY_BRANCH`.


It's also possible to run locust on your machine:

- install locust (look into .github/workflows/schedule-load-test.yml for the steps required to install it)
- Set the `FEEDS_AUTH_TOKEN` environment variable to contain the access token. You need to obtain your own token.
- Execute locust from the root folder of mobility-feed-api:
```
locust -f ./load-test/gtfs_user_test.py --host=https://api-qa.mobilitydatabase.org -u 100 -r 10 --headless -t 180 --only-summary --csv locust_result
```
  - This should generate 4 files in the current directory with names starting with locust_result
  - You might want to play with the -u, -r and -t parameters. See [locust command line options](https://docs.locust.io/en/stable/configuration.html)

Please make sure these boxes are checked before submitting your pull request - thanks!
